source/php/Entity/FieldSetter.php

```diff
@@ -39,4 +39,30 @@ public function setPostByParam(
             'assignment-eligibility'
         );
     }
+
+    /**
+     * Saves a media file for a post
+     *
+     * @param string $key          The key of the file in the $_FILES array.
+     * @param int    $postId       The ID of the post to attach the media file to.
+     * @param array  $allowedTypes An array of allowed file types.
+     *
+     * @return int|\WP_Error The attachment ID on success, or a \WP_Error object on failure.
+     */
+    public function savePostMedia(string $key, int $postId, array $allowedTypes = [])
+    {
+        if (!in_array($_FILES[$key]['type'], $allowedTypes)) {
+            $errorMessage = 'Invalid file type. Allowed types: ' . implode(', ', $allowedTypes);
+            return new \WP_Error('invalid_file_type', $errorMessage, ['status' => 400]);
+        }
```

Comment:
File type could be manipulated by client.
Might be better to use `wp_check_filetype` like:
```php
$fileType = wp_check_filetype(basename($_FILES[$key]['name']));
if (!in_array($fileType['type'], $allowedTypes)) {
    // ...
}
```

Think the code still works because of file type checks done by `media_handle_upload`.